apps/passport-server/src/util/telegramHelpers.ts

```diff
@@ -18,6 +36,53 @@ export interface SessionData {
 
 export type BotContext = Context & SessionFlavor<SessionData>;
 
+function isFulfilled<T>(
+  promiseSettledResult: PromiseSettledResult<T>
+): promiseSettledResult is PromiseFulfilledResult<T> {
+  return promiseSettledResult.status === "fulfilled";
+}
+
+/**
+ * Fetches the chat object for a list contaning a telegram chat id
+ */
+export const getEventsWithChats = async <
+  T extends LinkedPretixTelegramEvent | ChatIDWithEvents
+>(
+  db: Pool,
+  ctx: BotContext,
+  chats: T[]
+): Promise<EventWithChat<T>[]> => {
+  const eventsWithChatsRequests = chats.map(async (e) => {
+    return {
+      ...e,
+      chat: e.telegramChatID
+        ? ((await ctx.api.getChat(e.telegramChatID)) as TopicChat)
+        : null
+    };
+  });
+  const eventsWithChatsSettled = await Promise.allSettled(
+    eventsWithChatsRequests
+  );
+
+  const eventsWithChats = eventsWithChatsSettled
+    .filter(isFulfilled)
+    .map((e) => e.value);
+
+  return eventsWithChats;
+};
+
+export const findChatByEventIds = (
```

Comment:
This function is the magic of validating that a user's eventIds are associated with the same superevent (ie, Devconnect). TODO: Test this function.

Reply:
yep, we should test this

Reply:
i think you need to check if `eventIds` is empty. otherwise if it's empty, it will always return true on the first iteration of the for loop.

Reply:
You were right!